ui/src/message_popup/message_popup_page.jsx

```diff
@@ -133,33 +181,65 @@ export default React.createClass({
       questions,
       questionsAnswered,
       shouldShowStudentCards,
-      helpType
+      shouldShowSummary,
+      helpType,
+      sessionLength
     } = this.state;
     if (!hasStarted) return this.renderInstructions();
-    if (questionsAnswered >= questions.length) return this.renderDone();
+    if (questionsAnswered >= sessionLength) return this.renderDone();
 
     const question = this.state.questions[questionsAnswered];
     return (
       <div style={styles.container}>
+        <LinearProgress mode="determinate" value={questionsAnswered} max={sessionLength} />
         <VelocityTransitionGroup enter={{animation: "slideDown"}} leave={{animation: "slideUp"}} runOnMount={true}>
           <PopupQuestion
             key={JSON.stringify(question)}
             question={question}
             shouldShowStudentCard={shouldShowStudentCards}
+            shouldShowSummary={shouldShowSummary}
             helpType={helpType}
             limitMs={90000}
             onLog={this.onLog}
-            onDone={this.onQuestionDone}/>
+            onDone={this.onQuestionDone}
+            addResponseTime={this.addResponseTime}
```

Comment:
This seems too low-level to pass to the `PopupQuestion`, could this component just require that `PopupQuestion` pass back the time the question took, and then handle doing this within `onQuestionDone`?  Let me know if it's helpful to talk it out in person too!